src/s.ml

```diff
@@ -8,7 +8,8 @@ module type IO = sig
   type 'a stream
   type stream_count
 
-  val connect : string -> int -> fd t
+  val getaddrinfo : string -> string -> Unix.getaddrinfo_option list -> Unix.addr_info list t
+  val connect : Unix.socket_domain -> Unix.sockaddr -> fd t
```

Comment:
maybe we keep the old `connect` and add a `connect_full` or `connect_socket` function?

Reply:
My assumption was that this signature is only used internally, and therefore there's no need to maintain backward compatibility.

Since I've now moved the logic which calls getaddrinfo out of the IO modules to the client, I'm not sure how to move it back without duplicating the logic.

As I see it, it makes sense to keep the API surface of the IO modules as small as possible to reduce the amount of work when editing or adding new IO modules.